solver_options:
    regularization: 1e-9
    back_end: osqp
    front_end: ihqp
    nhqp_min_sv_ratio: 0.05


stack:
    - 
      - Rolling_FL
      - Rolling_FR
      - Rolling_HL
      - Rolling_HR
      - Postural
      - Wheel1
      - Wheel2
      - Wheel3
      - Wheel4

constraints: 
  - JointLimits
  - VelocityLimits
  - Steering_FL
  - Steering_FR
  - Steering_HL
  - Steering_HR
  # - Collision


JointLimits:
    type: JointLimits

VelocityLimits:
    type: VelocityLimits
    bound_scaling: 0.5
    limits:
      VIRTUALJOINT_1: 0.0
      VIRTUALJOINT_2: 0.0
      VIRTUALJOINT_4: 0.0
      VIRTUALJOINT_5: 0.0
      VIRTUALJOINT_6: 0.0


Collision:
    type: CollisionConstraint
    lib_name: libcartesio_collision_support.so
    bound_scaling: 0.1
    distance_threshold: 0.02
    max_pairs: 30
    collision_urdf_path: $(rospack find centauro_urdf)/urdf/centauro_capsule.urdf
    collision_srdf_path: $(rospack find centauro_srdf)/srdf/centauro_capsule.srdf

Steering_FL:
    type: CentauroSteering
    wheel_name: wheel_1
    weight: 100.0
    lib_name: libcentauro_cartesio_addon.so
    lambda: 0.1

Steering_FR:
    type: CentauroSteering
    wheel_name: wheel_2
    weight: 100.0
    lib_name: libcentauro_cartesio_addon.so
    lambda: 0.1

Steering_HL:
    type: CentauroSteering
    wheel_name: wheel_3
    weight: 100.0
    lib_name: libcentauro_cartesio_addon.so
    lambda: 0.1

Steering_HR:
    type: CentauroSteering
    wheel_name: wheel_4
    weight: 100.0
    lib_name: libcentauro_cartesio_addon.so
    lambda: 0.1

Rolling_FL:
    type: WheelRolling
    wheel_name: wheel_1
    lib_name: libcentauro_cartesio_addon.so
    wheel_radius: 0.078
    weight: 1.0

Rolling_FR:
    type: WheelRolling
    wheel_name: wheel_2
    lib_name: libcentauro_cartesio_addon.so
    wheel_radius: 0.078
    weight: 1.0

Rolling_HL:
    type: WheelRolling
    wheel_name: wheel_3
    lib_name: libcentauro_cartesio_addon.so
    wheel_radius: 0.078
    weight: 1.0

Rolling_HR:
    type: WheelRolling
    wheel_name: wheel_4
    lib_name: libcentauro_cartesio_addon.so
    wheel_radius: 0.078
    weight: 1.0

Wheel1:
    type: Cartesian
    distal_link: wheel_1
    indices: [2]

Wheel2:
    type: Cartesian
    distal_link: wheel_2
    indices: [2]

Wheel3:
    type: Cartesian
    distal_link: wheel_3
    indices: [2]

Wheel4:
    type: Cartesian
    distal_link: wheel_4
    indices: [2]

Postural:
    type: Postural
    lambda: 0.01
    weight: 0.1
    disabled_joints:
#      - VIRTUALJOINT_1
#      - VIRTUALJOINT_2
      - VIRTUALJOINT_3
#      - VIRTUALJOINT_4
#      - VIRTUALJOINT_5
#      - VIRTUALJOINT_6
      - ankle_yaw_1
      - ankle_yaw_2
      - ankle_yaw_3
      - ankle_yaw_4
      - j_wheel_1
      - j_wheel_2
      - j_wheel_3
      - j_wheel_4
